depends/toolchain.cmake.in

```diff
@@ -146,3 +146,12 @@ endif()
 if(NOT HARDENING AND "@no_harden@" STREQUAL "1")
   set(HARDENING OFF CACHE STRING "")
 endif()
+
+if("@multiprocess@" STREQUAL "1")
+  if(NOT MULTIPROCESS)
+    set(MULTIPROCESS ON CACHE STRING "")
+  endif()
+  if(NOT MPGEN_PREFIX)
+    set(MPGEN_PREFIX "${CMAKE_FIND_ROOT_PATH}/native" CACHE PATH "")
```

Comment:
I don't really understand what this is doing, might be helpful to add a comment. Particularly what is the last "" argument?

EDIT: I guess last "" arguments are for documentation? Are these arguments actually required? If they are it seems like it would be helpful to change them to something like "Set by depends" to document where they are coming from

Reply:
This command uses the following [signature](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry):
```cmake
set(<variable> <value>... CACHE <type> <docstring> [FORCE])
```
where the `<docstring>` is mandatory. It allows the user to override its value with `-DMPGEN_PREFIX=...` in the command line.

What about "libmultiprocess codegen tool prefix"?

Reply:
> What about "libmultiprocess codegen tool prefix"?

Well "libmultiprocess codegen tool prefix" already seems to be the documentation string for this option in `cmake/optional.cmake` so that seems very good.

But I'm confused about why any documentation string is required at all in `depends/toolchain.cmake.in` which should just be overriding values, not defining new options. If I understand https://cmake.org/cmake/help/latest/command/set.html correctly, in-memory values naturally override cached values, so it seems like the ideal thing would be to stop specifying CACHE \<type> \<docstring> arguments in this file, and just set the values in memory. If writing the cache here really is necessary though, I'd think appropriate doc string might be be something like "Overridden by depends build" to indicate that cached values have been overridden.

Reply:
> If writing the cache here really is necessary though...

It is. Otherwise, the user won't be able to override a build option via command line.
For example, passing `-DMULTIPROCESS=OFF` won't have any effect.

This is a specific of toolchain files.

> ... I'd think appropriate doc string might be be something like "Overridden by depends build" to indicate that cached values have been overridden.

Docstrings are available via `-LH` command-line option or in `cmake-gui`. And you are right that they shouldn't be empty.

However, its content must just be copy-pasted from the `option()` commands from the main `CMakeLists.txt` file. Otherwise, for example, when the user passes `-DMULTIPROCESS=OFF`, it ends with:
```
// Overridden by depends build
MULTIPROCESS:STRING=OFF
```
which is misleading.

Reply:
> > ... I'd think appropriate doc string might be be something like "Overridden by depends build" to indicate that cached values have been overridden.
> 
> Docstrings are available via `-LH` command-line option or in `cmake-gui`. And you are right that they shouldn't be empty.

Fixed in https://github.com/hebasto/bitcoin/pull/122.